File: 프로그래머스/1/176963. 추억 점수/추억 점수.py
def solution(name, yearning, photo):
    answer = []
    for ph in photo:
        tmp = 0
        for ph_name in ph:
            if ph_name not in name:
                continue
            idx = name.index(ph_name)
            tmp += yearning[idx]
        answer.append(tmp)
    return answer

File: 프로그래머스/1/176963. 추억 점수/추억 점수.java
class Solution {
    public int[] solution(String[] name, int[] yearning, String[][] photo) {
        int[] answer = new int[photo.length];
        int idx = 0;
        for(String[] p : photo){
            int score = 0;
            for(String n : p){
                int i = 0;
                for(; i < name.length; i++){
                    if(n.equals(name[i])){
                        break;
                    }
                }
                if(i == name.length)
                    continue;
                else{
                    score += yearning[i];
                }
            }
            answer[idx++] = score;
        }
        return answer;
    }
}